### PR TITLE
[vcpkg-tools] update cmake and git (windows only)

### PR DESCRIPTION
**Pull request by Neumann-A**
Just update cmake to 3.22.2 and git on windows

supersedes #21456 
~~depends #22998~~
~~depends #22846~~
~~depends #23010~~

I'll cleanup this PR when the dependent PRs have been merged.

### CODE REVIEW

**Comment by JackBoosY** [comment]
Related: https://github.com/microsoft/vcpkg/pull/21456.

File: ports/qtbase/portfile.cmake [hunk 1]
```diff
@@ -130,6 +129,12 @@ vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_GUI_OPTIONS
     "jpeg"                FEATURE_jpeg
     "png"                 FEATURE_png
     #"opengl"              INPUT_opengl=something
+    "xlib"                FEATURE_xlib
+    "xkb"                 FEATURE_xkb
+    "xcb"                 FEATURE_xcb
+    "xcb-xlib"            FEATURE_xcb_xlib
+    "xkbcommon-x11"       FEATURE_xkbcommon_x11
+    "xrender"             FEATURE_xrender
     INVERTED_FEATURES
```
Comment (Neumann-A) [inline ports/qtbase/portfile.cmake:138]: I expect the Qt 6.2.3 PR to be merged before this one. This is just to make CI pass

**Comment by Neumann-A** [comment]
hmm linux qtbase fails with:

```
ERROR: Feature "xkbcommon_x11": Forcing to "ON" breaks its condition:
    QT_FEATURE_xkbcommon AND XKB_COMMON_X11_FOUND
Condition values dump:
    QT_FEATURE_xkbcommon = "ON"
    XKB_COMMON_X11_FOUND = "0"
```
the search call is:
`qt_find_package(XKB_COMMON_X11 0.5.0 PROVIDED_TARGETS PkgConfig::XKB_COMMON_X11 MODULE_NAME gui QMAKE_LIB xkbcommon_x11)`
 provision script has `libxkbcommon-x11-dev` so it should be installed.

`pkg_check_modules(XKB_COMMON_X11 "xkbcommon-x11>=0.4.1" IMPORTED_TARGET)`

**Comment by Neumann-A** [comment]
So the problem boils down that qt and hidapi uses pkgconf build by vcpkg which is not setup to find system dependencies which is why the pkg_search_module call is failing for bot of them.

**Comment by Neumann-A** [comment]
failing:
- `qtinterfaceframework:arm64-windows` due to pkgconf build for arm found but can of course not run on x64
- cspice:x64-uwp due to header conflict (baseline regression)

**Comment by JackBoosY** [comment]
Wow, thanks!
Since this PR was merged, can we continue https://github.com/microsoft/vcpkg/pull/17487 ?

**Comment by dg0yt** [comment]
> Since this PR was merged, can we continue #17487 ?

Why continue `pkg-config`?  There is `pkgconf`.

**Pull request merged**